content/doc/developer/forms/form-validation.adoc

```diff
@@ -45,6 +45,21 @@ public FormValidation doCheckThreads(@QueryParameter("value") final String meani
 
 This naming convention is how Jenkins wires up your form validation method. The return type also must be FormValidation, which offers various static factory methods to create an instance. Form validation can be of 3 different kinds: `ok`, `warning` or `error`. Validation message can either be represented in plaintext or use HTML markup. Several validations of different kinds can be reported wrapping then into one using aggregate factory method.
 
+=== Setting up form validation within a Jelly template
+
+To configure form validation within a Jelly template to call `doCheckXxx` methods, defined in a particular class in Jenkins, you need to utilize the `checkUrl` and `checkDependsOn` attributes.
+Here's a breakdown of how these attributes are used:
+
+The `checkUrl` attribute specifies the URL that Jenkins requests to perform form validation.
+Typically, this URL points to the Jenkins server where it handles the validation request.
+The value entered in the input field is checked via AJAX against this URL and any errors encountered are rendered under the text field.
+This URL is associated with the method responsible for performing the validation (`doCheckXxx` method) within your Descriptor class.
+
+The `checkDependsOn` attribute defines the dependency between the field being validated and the field(s) whose values are required for validation.
+It specifies the ID(s) of the field(s) that the validation depends on.
+When the dependent field(s) change, Jenkins triggers the validation of the current field.
+Additionally, the fields specified in `checkDependsOn` are sent along with the form validation request.
```

Comment:
I am not sure if this is accurate, since it makes it sounds like the tag along is intentional and deliberate. 

For more info, please see https://www.jenkins.io/doc/developer/security/csp/#legacy-javascript-checkurl-validation

Reply:
Bacially the `checkUrl` mode requires us to set the `checkDependsOn` attribute, but it can be simply an empty string.

Reply:
Also, what do you mean by "When the dependent field(s) change"?

Reply:
It sounds like they change by themselves without any developer intervention to me 😂

Reply:
> since it makes it sounds like the tag along is intentional and deliberate.

But it is deliberate.

> When the dependent field(s) change,

Actually the current field is *dependent* on the fields specified in `checkDependsOn`, not the other way around.

Reply:
Yeah, I guess I just don't like the wording.

Reply:
Okay got it sir, sorry for any confusion